lib/agent/actions/fileretrieval/files_storage.js

```diff
@@ -9,41 +9,42 @@ var join      = require('path').join,
 var logger   = common.logger;
 var watching  = false;
 
-var files_path = join(common.system.paths.config, 'files.db');
-storage.init(files_path);
+// var files_path = join(common.system.paths.config, 'files.db');
+// storage.init(files_path);
 
-
-exports.store = function(id, path, user) {
-  if (!user) return;
-  logger.debug('Storing file_id in DB: ' + id);
-  var opts = {
-    path: path,
-    user: user
-  }
-  storage.set(id, opts);
-}
-
-exports.del = function(id) {
-  logger.debug('Removing file id from DB: ' + id);
-  storage.del(id);
-}
-
-exports.show = function() {
+var exist = function(id, cb) {
+  var key = "fr/"+id
   storage.all(function(err, files) {
     if (err)
       return logger.error(err.message);
+    if (files[key]) 
+      return cb(true);
+    else
+      return cb(false);
+  })
+}
 
-    var count = Object.keys(files).length;
-    if (count <= 0)
+exports.store = function(id, path, size, user) {
+  exist(id, function(cb) {
+    if (cb == false) {
+      logger.debug('Storing file_id in DB: ' + id);
+      var opts = {
+        path: path,
+        size: size,
+        user: user
+      }
+      storage.set("fr/"+id, opts);
```

Comment:
I would use a constant such as `NAMESPACE` within this file, so that you call storage methods like `storage.set(NAMESPACE + id)` where `NAMESPACE` is `fr/`